### PR TITLE
[PUX-2739] | Add iOS UI color customisation

### PR DESCRIPTION
**Pull request by TheInkedEngineer**
This PR adds the possibility to customize the colors in the UI of the SDK

Example:

```
const iOSTheme = {
  lightColors: {
    backgroundSecondary: "#123A2B"
  },
  darkColors: {
    backgroundPrimary: "#000000"
  },
  fontFamilyName: "Kanit"
}

const theme = {
  ios: iOSTheme
}
```
Output with a custom font:

### CODE REVIEW

**Comment by tl-keenen-charles** [comment]
Looks good to me. But the readme will need some updates since right now there's an Android section above this and the example covers both platforms now. I'd also update the bit below about supported color formats if it's different on iOS.

File: DemoApp/ios/Podfile [hunk 1]
```diff
@@ -10,6 +10,8 @@ target 'DemoApp' do
   # Flags change depending on the env values.
   flags = get_default_flags()
 
+  pod 'TrueLayerPaymentsSDK', :git => 'https://github.com/TrueLayer/TrueLayer-iOS-SDK', :branch => 'release/2.3.0'
```
[the remark this answers is not in the document]
Comment (tl-lukasz-franieczek) [reply DemoApp/ios/Podfile:13]: I think the idea is that those files will be updated with each release. The same situation will be there for Android gradle file.

[the remark this answers is not in the document]
Comment (TheInkedEngineer) [reply DemoApp/ios/Podfile:13]: I left some comments of my own, but they were left pending... 😮‍💨 

Screen Shot 2023-02-17 at 09 09 29

File: RNTrueLayerPaymentsSDK/js/models/theme/Theme.ts [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
+import { iOSColors } from "./iOSColors";
 
-export type Theme = {
-  android?: {
-    lightColors?: AndroidColors;
-    darkColors?: AndroidColors;
-    typography?: AndroidTypography;
+export type Theme = Partial<{
+  android: {
+    lightColors: AndroidColors;
```
[the remark this answers is not in the document]
Comment (TheInkedEngineer) [reply RNTrueLayerPaymentsSDK/js/models/theme/Theme.ts:7]: So final outcome would be:

```
export type Theme = Partial<{
  android: Partial<{
    lightColors?: AndroidColors;
    darkColors?: AndroidColors;
    typography?: AndroidTypography;
    }>
  }>
```

Right, @enricocasadei?

[the remark this answers is not in the document]
Comment (enricocasadei) [reply RNTrueLayerPaymentsSDK/js/models/theme/Theme.ts:7]: without the question mark 💪

File: docs/DOCUMENTATION.md [hunk 3]
```diff
@@ -211,4 +211,41 @@ TrueLayerPaymentsSDKWrapper.configure(Environment.Sandbox, theme).then(
...
+  },
+  darkColors: {
+    contentSecondary: "#ABABAB",
+    ...
+  },
+  fontFamilyName: "Kanit"
+}
+
+const theme =  {
+  android: androidTheme,
```
Comment (tl-daven-gomes) [inline docs/DOCUMENTATION.md:233]:
```suggestion
  ios: iOSTheme,
```

**Review by tl-daven-gomes — APPROVED** [review]

File: docs/DOCUMENTATION.md (hunk 3 above)
Comment (TheInkedEngineer) [reply docs/DOCUMENTATION.md:233]: Good catch

**Review by tl-lukasz-franieczek — APPROVED** [review]

**Pull request merged**